pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -0,0 +1,864 @@
+		for _, wl := range a.byPod {
+			if !(wl.Namespace == sa.Namespace && wl.ServiceAccount == sa.Name) {
+				continue
+			}
+			wl := &model.WorkloadInfo{Workload: proto.Clone(wl).(*workloadapi.Workload)}
+			addrs := make([][]byte, 0, len(wl.WaypointAddresses))
+			filtered := false
+			for _, a := range wl.WaypointAddresses {
+				if !bytes.Equal(a, addr) {
+					addrs = append(addrs, a)
+				} else {
+					filtered = true
+				}
+			}
+			wl.WaypointAddresses = addrs
+			if filtered {
+				// If there was a change, also update the VIPs and record for a push
+				updates[model.ConfigKey{Kind: kind.Address, Name: wl.ResourceName()}] = struct{}{}
+				a.byPod[wl.ResourceName()] = wl
+				for vip := range wl.VirtualIps {
+					a.insertWorkloadToService(vip, wl)
+				}
+			}
+		}
+	} else {
+		for _, wl := range a.byPod {
+			if !(wl.Namespace == sa.Namespace && wl.ServiceAccount == sa.Name) {
+				continue
+			}
+			found := false
+			for _, a := range wl.WaypointAddresses {
+				if bytes.Equal(a, addr) {
+					found = true
+					break
+				}
+			}
+			if !found {
+				wl := &model.WorkloadInfo{Workload: proto.Clone(wl).(*workloadapi.Workload)}
+				wl.WaypointAddresses = append(wl.WaypointAddresses, addr)
+				// If there was a change, also update the VIPs and record for a push
+				updates[model.ConfigKey{Kind: kind.Address, Name: wl.ResourceName()}] = struct{}{}
+				a.byPod[wl.ResourceName()] = wl
+				for vip := range wl.VirtualIps {
+					a.insertWorkloadToService(vip, wl)
+				}
+			}
+		}
+	}
+	return updates
+}
+
+// All return all known workloads. Result is un-ordered
+func (a *AmbientIndex) All() []*model.WorkloadInfo {
+	a.mu.RLock()
+	defer a.mu.RUnlock()
+	res := make([]*model.WorkloadInfo, 0, len(a.byPod))
+	// byPod will not have any duplicates, so we can just iterate over that.
+	for _, wl := range a.byPod {
+		res = append(res, wl)
+	}
+	return res
+}
+
+func (c *Controller) Policies(requested sets.Set[model.ConfigKey]) []*workloadapi.RBAC {
+	cfgs, err := c.configController.List(gvk.AuthorizationPolicy, metav1.NamespaceAll)
+	if err != nil {
+		// TODO: handle this somehow?
+		return nil
+	}
+	l := len(cfgs)
+	if len(requested) > 0 {
+		l = len(requested)
+	}
+	res := make([]*workloadapi.RBAC, 0, l)
+	for _, cfg := range cfgs {
+		k := model.ConfigKey{
+			Kind:      kind.AuthorizationPolicy,
+			Name:      cfg.Name,
+			Namespace: cfg.Namespace,
+		}
+		if len(requested) > 0 && !requested.Contains(k) {
+			continue
+		}
+		pol := convertAuthorizationPolicy(c.meshWatcher.Mesh().GetRootNamespace(), cfg)
+		if pol == nil {
+			continue
+		}
+		res = append(res, pol)
+	}
+	return res
+}
+
+func isNil(v interface{}) bool {
+	return v == nil || (reflect.ValueOf(v).Kind() == reflect.Ptr && reflect.ValueOf(v).IsNil())
+}
+
+func (c *Controller) selectorAuthorizationPolicies(ns string, lbls map[string]string) []string {
+	// Since this is an interface a normal nil check doesn't work (...)
+	if isNil(c.configController) {
```

Comment:
I'm confused what could be stored here that would fail this check?

Reply:
its not set for non-config clusters